src/main/resources/net/uaznia/lukanus/hudson/plugins/gitparameter/javascript/git-parameter-select.js

```diff
@@ -4,74 +4,99 @@ function gitParameterUpdateSelect(listBox, url, divId, config) {
         status.innerHTML = "";
     }
     config.onSuccess = function (rsp) {
-        l.removeClassName("select-ajax-pending");
+        l.classList.remove("select-ajax-pending");
         var currentSelection = l.value;
 
         // clear the contents
         while (l.length > 0) l.options[0] = null;
 
         var selectionSet = false; // is the selection forced by the server?
         var possibleIndex = null; // if there's a new option that matches the current value, remember its index
-        var opts = eval('(' + rsp.responseText + ')').values;
-        for (var i = 0; i < opts.length; i++) {
-            l.options[i] = new Option(opts[i].name, opts[i].value);
-            if (opts[i].selected) {
-                l.selectedIndex = i;
-                selectionSet = true;
+        rsp.json().then(json => {
+            var opts = json.values;
+            for (var i = 0; i < opts.length; i++) {
+                l.options[i] = new Option(opts[i].name, opts[i].value);
+                if (opts[i].selected) {
+                    l.selectedIndex = i;
+                    selectionSet = true;
+                }
+                if (opts[i].value === currentSelection)
+                    possibleIndex = i;
             }
-            if (opts[i].value === currentSelection)
-                possibleIndex = i;
-        }
 
-        // if no value is explicitly selected by the server, try to select the same value
-        if (!selectionSet && possibleIndex != null)
-            l.selectedIndex = possibleIndex;
+            // if no value is explicitly selected by the server, try to select the same value
+            if (!selectionSet && possibleIndex != null)
+                l.selectedIndex = possibleIndex;
 
-        if (originalOnSuccess !== undefined)
-            originalOnSuccess(rsp);
+            if (originalOnSuccess !== undefined)
+                originalOnSuccess(rsp);
 
-        var errors = eval('(' + rsp.responseText + ')').errors
-        let error_div = $("git_parameter_errors_" + divId);
-        if (errors.length !== 0) {
-            error_div.show();
-            error_div.addClassName("error");
-            let $error_ul = $("git_parameter_errors_ul_" + divId);
-            var lis = "";
-            for (var j = 0; j < errors.length; j++) {
-                lis += "<li>" + escapeHTML(errors[j]) + "</li>"
+            var errors = json.errors
+            let error_div = document.getElementById("git_parameter_errors_" + divId);
+            if (errors.length !== 0) {
+                error_div.style.display = "";
+                error_div.classList.add("error");
+                let $error_ul = document.getElementById("git_parameter_errors_ul_" + divId);
+                var lis = "";
+                for (var j = 0; j < errors.length; j++) {
+                    lis += "<li>" + escapeHTML(errors[j]) + "</li>"
+                }
+                $error_ul.update(lis);
+            } else {
```

Comment:
Update is a Prototype function that is not present in native JavaScript. When I exercise this code, I get the following exception in the console log:

```
Uncaught (in promise) TypeError: $error_ul.update is not a function
    onSuccess http://127.0.0.1/adjuncts/a2c26138/net/uaznia/lukanus/hudson/plugins/gitparameter/javascript/git-parameter-select.js:52
    promise callback*gitParameterUpdateSelect/config.onSuccess http://127.0.0.1/adjuncts/a2c26138/net/uaznia/lukanus/hudson/plugins/gitparameter/javascript/git-parameter-select.js:23
    gitParameterUpdateSelect http://127.0.0.1/adjuncts/a2c26138/net/uaznia/lukanus/hudson/plugins/gitparameter/javascript/git-parameter-select.js:84
    promise callback*gitParameterUpdateSelect http://127.0.0.1/adjuncts/a2c26138/net/uaznia/lukanus/hudson/plugins/gitparameter/javascript/git-parameter-select.js:82
    <anonymous> http://127.0.0.1/adjuncts/a2c26138/net/uaznia/lukanus/hudson/plugins/gitparameter/javascript/git-parameter-select.js:122
    h http://127.0.0.1/static/a2c26138/scripts/hudson-behavior.js:1937
    refillOnChange http://127.0.0.1/static/a2c26138/scripts/hudson-behavior.js:1961
    <anonymous> http://127.0.0.1/adjuncts/a2c26138/net/uaznia/lukanus/hudson/plugins/gitparameter/javascript/git-parameter-select.js:120
    applySubtree http://127.0.0.1/static/a2c26138/scripts/behavior.js:135
    applySubtree http://127.0.0.1/static/a2c26138/scripts/behavior.js:126
    applySubtree http://127.0.0.1/static/a2c26138/scripts/behavior.js:107
    apply http://127.0.0.1/static/a2c26138/scripts/behavior.js:90
    start http://127.0.0.1/static/a2c26138/scripts/behavior.js:85
    onload http://127.0.0.1/static/a2c26138/scripts/behavior.js:152
    onload http://127.0.0.1/static/a2c26138/scripts/behavior.js:152
    addLoadEvent http://127.0.0.1/static/a2c26138/scripts/behavior.js:151
    createSearchBox http://127.0.0.1/static/a2c26138/scripts/hudson-behavior.js:2420
    <anonymous> http://127.0.0.1/adjuncts/a2c26138/jenkins/views/JenkinsHeader/search-box.js:4
    <anonymous> http://127.0.0.1/adjuncts/a2c26138/jenkins/views/JenkinsHeader/search-box.js:6
```

Reply:
Thanks! This looks like it can be replaced by `.innerHTML = `